colcon_bundle/verb/_overlay_utilities.py

```diff
@@ -50,15 +79,35 @@ def create_dependencies_overlay(staging_path, overlay_path):
     """
     dependencies_staging_path = staging_path
     dependencies_tar_gz_path = overlay_path
-
     logger.info('Dependencies changed, updating {}'.format(
         dependencies_tar_gz_path
     ))
+    context_vars_sh = {'shell': 'sh'}
+    context_vars_bash = {'shell': 'bash'}
+
+    generate_template(
+        'v2_setup.sh.jinja2',
+        'v2_setup_template.sh',
+        context_vars_sh
+    )
+
+    generate_template(
+        'v2_setup.sh.jinja2',
+        'v2_setup_template.bash',
+        context_vars_bash
+    )
+
     assets_directory = os.path.join(
         os.path.dirname(os.path.realpath(__file__)), 'assets')
-    shellscript_path = os.path.join(assets_directory, 'v2_setup.sh')
+    shellscript_path = os.path.join(assets_directory, 'v2_setup_template.sh')
```

Comment:
It's quite common to suffix the files with the templating engine.

See Open Robotics code there : https://gitlab.uni-koblenz.de/robbie/ros_buildfarm/-/tree/2c97c32f414341ab9ab7a2d55d55de4c7f2793c9/ros_buildfarm/templates/doc

It's em because they use empy: https://pypi.org/project/empy/

In our case, we could call those files e.g. "v2_setup.jinja2.sh"

Reply:
The template is called v2_setup.sh.jinja2. This is just creating the actual setup file from the template to be bundled.